Meta/Lagom/CMakeLists.txt

```diff
@@ -343,9 +343,12 @@ install(TARGETS LibC LibCrypt LibSystem NoCoverage EXPORT LagomTargets)
 # AK
 add_serenity_subdirectory(AK)
 lagom_lib(AK ak SOURCES ${AK_SOURCES})
-if (${CMAKE_SYSTEM_NAME} MATCHES "BSD$" OR HAIKU)
-    # BSD Platforms and Haiku have backtrace(3) in a separate library
-    target_link_libraries(AK PRIVATE execinfo)
+find_package(Backtrace)
+if( Backtrace_FOUND AND NOT "${Backtrace_LIBRARIES}" STREQUAL "" )
+    # Non-glibc systems have backtrace(3) in a separate library
+    get_filename_component(Backtrace_LIBFILE ${Backtrace_LIBRARIES} NAME_WE)
+    string(REGEX REPLACE "^lib" "" Backtrace_LIBFILE ${Backtrace_LIBFILE})
+    target_link_libraries(AK PRIVATE ${Backtrace_LIBFILE})
 endif()
```

Comment:
The way I've done it is directly inspired by how the llvm project does it in one of their files:
```cmake
if( HAVE_BACKTRACE AND NOT "${Backtrace_LIBRARIES}" STREQUAL "" )
  # On BSDs, CMake returns a fully qualified path to the backtrace library.
  # We need to remove the path and the 'lib' prefix, to make it look like a
  # regular short library name, suitable for appending to a -l link flag.
  get_filename_component(Backtrace_LIBFILE ${Backtrace_LIBRARIES} NAME_WE)
  STRING(REGEX REPLACE "^lib" "" Backtrace_LIBFILE ${Backtrace_LIBFILE})
  set(system_libs ${system_libs} ${Backtrace_LIBFILE})
endif()
```
Reading the documentation for `target_link_libraries` I now realize that in this case this is unneeded as it can take a full path to a library file no problem.
It's an oversight on my part.

As for the check in the if statement, on glibc systems `Backtrace_FOUND` is set to true but `Backtrace_LIBRARIES` is left empty.

Also, Thank you for mentioning https://grep.app/! I hadn't heard of it before.

Reply:
Great! fwiw, LLVM's CMake has quite a lot of ... old.. decisions  in it.  I'm not sure that I'd use it as a reference for "modern CMake".